cmd/config.go

```diff
@@ -254,9 +259,78 @@ func validateConfig(conf Config, isExecutable func(string) bool) error {
 		}
 	}
 
+	// If there are thresholds to validate, the registry paramater is not allowed to be nil.
+	// Note that the reason for passing it as a pointer in the first place is
+	// because it holds a Mutex, which effectively forbids passing it by value.
+	if conf.Thresholds != nil && len(conf.Thresholds) > 0 && registry == nil {
+		err := fmt.Errorf(
+			"unable to validate thresholds configuration; " +
+				"reason: provided registry is nil",
+		)
+		errList = append(errList, err)
+		return consolidateErrorMessage(errList, "there were problems while validating the specified script configuration: ")
+	}
+
+	for thresholdName, thresholds := range conf.Thresholds {
+		// Fetch metric matching the threshold's name
+		metric, ok := registry.Get(thresholdName)
+		if !ok {
```

Comment:
This will also break [this](https://github.com/grafana/k6/issues/2390#issuecomment-1041407562) example even though in practice all the thresholds are *valid*, they just should be evaluated differently from how they are currently. 

The problem here is that `thresholdName` includes the tags while we can only check the *name* of the treshold not the name+ the tags. This is the error:

> ERRO[0000] invalid threshold defined on http_reqs{key:""}; reason: no metric named http_reqs{key:""} found


p.s. The thresholds also are evaluated wrongly with this PR as well - a tag with empty value in threshold is always "matched" even if it doesn't have the tag at all

Reply:
Thanks for pointing the issue out. Do you have pointers or a proposal to make this work, or improve it?

Reply:
:shrug: I guess something a kin to https://github.com/grafana/k6/blob/737dfa7ced6e4f94dbb3d6f2c2837085e783ac09/core/engine.go#L106-L114